File: Basic Programming/operators/going_to_office.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{ long long d, oc,of,od,cs,cb,cm;
long long cd;
long cost_online,cost_offline;

 cin>>d;
 cin>>oc>>of>>od;
 cin>>cs>>cb>>cm>>cd;

 cost_online=  oc+(d-of)*(od)  ;
 cost_offline= (cb+((d/cs)*(cm))+((d)*(cd)));


 if(cost_offline>=cost_online)
 cout<<"Online Taxi";
 else
 cout<<"Classic Taxi";


}

File: Basic Programming/operators/going_to_office.py
distance = int(input())
o_c, o_f, o_d = [int(i) for i in input().split(' ')]
c_s, c_b, c_m, c_d = [int(j) for j in input().split(' ')]

def ceildiv(a, b):
    return -(-a // b)

def online_cost(d:int):
    if d <= o_f:
        return o_c
    elif d > o_f:
        return o_c + o_d*(d-o_f)

def classic_cost(d:int):
    return c_b + ceildiv(d, c_s)*c_m + d*c_d

if online_cost(distance) <= classic_cost(distance):
    print('Online Taxi')
elif online_cost(distance) > classic_cost(distance):
    print('Classic Taxi')
